File: cpp/NumberGuess.cpp
#include <iostream>
#include <ctime>
#include <cstdlib>

using namespace std;

int main(){
  cout << "Welcome to Number Guessing Game" << endl;
  srand(time(NULL)); // seeding random number generator
  int my_number = rand() % 100; // Generate random number
  int guess = 0;
  cout << "I am thinking of a number between 1 and 100 " << endl;
  while (guess != my_number){
    cout << "Guess my number:";
    cin >> guess;
    if (guess == my_number){
      cout << "\nYou Guessed it!!!"<< endl;
      break;
    }else if (guess > my_number){
      cout << "Lower ..." << endl;
    }else{
      cout << "Higher ..." << endl;
    }
  }
  return 0;
}

File: javascript/number_guess.js
function check_number(){
  var guess = parseInt($("#guess").val());
  if (guess == window.thinking_number){
    update_display("You Gussed it");
  }else if(guess > window.thinking_number){
    update_display("Lower... Guess Again");
  }else{
    update_display("Higher... Guess Again");
  }
}
function update_display(message){
  $("#response h2").html(message);
}

function init(){
 window.thinking_number = Math.floor(Math.random()*101);
 update_display("I am thinking of a number between 1 and 100");
}

$(document).ready(function(){
  
  init();
  $("#go").bind("click", check_number);
})